pkg/reconciler/autoscaling/hpa/resources/hpa.go

```diff
@@ -66,6 +70,19 @@ func MakeHPA(pa *v1alpha1.PodAutoscaler, config *autoscalerconfig.Config) *autos
 				},
 			}}
 		}
+
+	case autoscaling.Memory:
+		if target, ok := pa.Target(); ok {
+			memory := resource.NewQuantity(int64(target)*1024*1024, resource.BinarySI)
```

Comment:
In K8s the typical suffixes are listed [here](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/), check `Mi` vs `MiB`. Any reason why we dont support them in the definition and Mib is used by default?

Reply:
We cannot because  `target` is defined as a `float64` currently. We'd have to change that to allow different representations, but Megabyte seems to be a reasonable default (modulo the diff you pointed out)

Reply:
I would vote to change that for better UX.

Reply:
Or we can support user to set suffixes by another annotation:`autoscaling.knative.dev/memory_suffix`(only takes effect when metrics is `memory`). Annotation key is one of `E, P, T, G, M, k` and `Ei, Pi, Ti, Gi, Mi, Ki`.

Reply:
sounds like it doesn't necessarily have to be part of this PR though and can be a subsequent enhancement.

My 2c: imo the float is fine :shrug:.

Reply:
Sure let's do it in another pr dont consider it a blocker or anything.

Reply:
yeah, I will track that subsequent enhancement in another PR